CMakeLists.txt

```diff
@@ -306,6 +314,15 @@ else()
 	endif (MSVC)
 endif()
 
+if (NOT MSVC)
+	if (abi STREQUAL "c++03")
+		add_definitions(-std=${abi})
```

Comment:
In practice, you likely get better compatibility assuming that a compiler is GCC compatible than only supporting known ones. ARM toolchain and ICC both use GCC basic syntax for these kinds of things, and it's easier to add exceptions for the cases where a certain compiler doesn't support a given option.

In general, I'm a fan of writing things in a generic way, but without an easy way to test it, I think it's reasonable to stick to GCC, Clang and MSVC as requirements for a build script change.